### PR TITLE
Added new files in C

### PR DESCRIPTION
**Pull request by YashovardhanSiramdas**
**Changes:**
Added six new files:-
1)Binomial Coefficient in C
2)Reverse factorial in C
3)Tribonacci numbers in C
4)AVL Tree in C
5)Median in C
6)Factorial in C

### CODE REVIEW

**Comment by YashovardhanSiramdas** [comment]
I've followed the C style guide for braces and indentations.Can you please point out in which file and where I didn't follow the style guide so I can rectify it?

**Comment by YashovardhanSiramdas** [comment]
I think I've made sure now the code follows C Style Guide.Kindly let me know if there is something else I need to do regarding this.

**Comment by OkarinRintaro** [comment]
There is
```C
int c=0,f=0;

int height(  

```
but necessary:  
```C
int c = 0, f = 0;

int   
height(
```
and etc.

All return statements should have the value wrapped in parenthesis.

**Comment by YashovardhanSiramdas** [comment]
Thanks for the suggestion **OkarinRintaro**. I've made the changes exactly according to Style Guide now.Let me now if this is okay.

File: code/data_structures/src/tree/binary_tree/avl_tree/AVL.c [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+{
+	struct node *jptr = rptr;
+	struct node *kptr = rptr->rcptr;
+	jptr->rcptr = RightRotate(kptr);
+
+	return LeftRotate(jptr);
+}
+
+struct node *
+insert(struct node *rptr,int x)
```
Comment (OkarinRintaro) [inline code/data_structures/src/tree/binary_tree/avl_tree/AVL.c]: :sparkles: Looks better.  
Put one space between function args.

Comment (YashovardhanSiramdas) [reply code/data_structures/src/tree/binary_tree/avl_tree/AVL.c]: Yeah,done. Does it look good now?

**Comment by YashovardhanSiramdas** [comment]
@arnavb @AdiChat  I've made changes to code according to Style Guide. Does it look good now?

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

**Pull request merged**